server/src/main/java/io/unitycatalog/server/persist/utils/UriUtils.java

```diff
@@ -0,0 +1,245 @@
+    }
+
+    return modelStorageRoot;
+  }
+
+  private static String getModelDirectoryURI(String entityFullName) {
+    return getModelStorageRoot() + "/" + entityFullName.replace(".", "/");
+  }
+
+  public static String getModelStorageLocation(String catalogId, String schemaId, String modelId) {
+    return getModelDirectoryURI(catalogId + "." + schemaId + ".models." + modelId);
+  }
+
+  public static String getModelVersionStorageLocation(
+      String catalogId, String schemaId, String modelId, String versionId) {
+    return getModelDirectoryURI(
+        catalogId + "." + schemaId + ".models." + modelId + ".versions." + versionId);
+  }
+
+  public static String createStorageLocationPath(String uri) {
+    return updateDirectoryFromUri(uri, Operation.CREATE, Optional.empty()).toString();
+  }
+
+  public static String deleteStorageLocationPath(String uri) {
+    return updateDirectoryFromUri(uri, Operation.DELETE, Optional.empty()).toString();
+  }
+
+  private static URI updateDirectoryFromUri(
+      String uri,
+      Operation op,
+      Optional<GenerateTemporaryModelVersionCredentialsResponse> credentials) {
+    URI parsedUri = URI.create(uri);
+    validateURI(parsedUri);
+    if (!parsedUri.getScheme().equals("file") && credentials.isEmpty()) {
+      throw new BaseException(
+          ErrorCode.INVALID_ARGUMENT, "Empty credentials passed for a non-file based URI");
+    }
+    try {
+      if (parsedUri.getScheme().equals("file")) {
+        return updateLocalDirectory(parsedUri, op);
+      } else if (parsedUri.getScheme().equals("s3")
+          && credentials.get().getAwsTempCredentials() != null) {
+        return updateS3Directory(parsedUri, op, credentials.get().getAwsTempCredentials());
+      } else if (parsedUri.getScheme().equals("gc")
+          && credentials.get().getGcpOauthToken() != null) {
+        return updateGcDirectory(parsedUri, op, credentials.get().getGcpOauthToken());
+      } else if (parsedUri.getScheme().equals("abfs")
+          && credentials.get().getAzureUserDelegationSas() != null) {
+        return updateAbsDirectory(parsedUri, op, credentials.get().getAzureUserDelegationSas());
+      }
+    } catch (Exception e) {
+      throw new BaseException(
+          ErrorCode.INTERNAL,
+          "Error attempting to "
+              + op.name()
+              + " URI "
+              + parsedUri.toString()
+              + ": "
+              + e.getMessage());
+    }
+    throw new BaseException(
+        ErrorCode.INVALID_ARGUMENT, "Unknown scheme detected: " + parsedUri.getScheme());
+  }
+
+  private static boolean isValidURI(String uri) {
+    try {
+      URI testURI = new URI(uri);
+      if (testURI.getScheme() != null && testURI.getPath() != null) {
+        return true;
+      }
+      return false;
+    } catch (URISyntaxException e) {
+      return false;
+    }
+  }
+
+  private static URI updateLocalDirectory(URI parsedUri, Operation op) throws IOException {
```

Comment:
Note to self : Difference from FileUtils : In FileUtils the input is `Path` object for both CREATE AND DELETE and the functions exist independently. 
In create we are returning `Path.toUri()` which is not required here as we already have the URI as input. 
Delete is not returning anything.